### PR TITLE
Replace base R message/warning/stop with `{cli}`

### LINKED ISSUE #546: Move to `{cli}` for messaging
One reason to do so:

https://github.com/epiforecasts/EpiNow2/pull/534#issuecomment-1943972293

### PR DESCRIPTION
**Pull request by jamesmbaazam**
## Description

This PR closes #546.

This PR replaces the use of base condition signalling with those provided by `{cli}` as it offers more powerful features such as the ability to display a warning every few hours.

As part of this PR, I've updated the `.lintr` config file to catch future uses of base messaging. This was really useful in the refactoring exercise as it caught all occurrences. There were some false positives in the `regional_epinow.R`, so I embraced them in `#nolint undesirable_function_linter` tags to counter the configs I've set up. 

## Initial submission checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [x] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [x] I have added a news item linked to this PR.

## After the initial Pull Request 

- [ ] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: .lintr [hunk 1]
```diff
@@ -2,7 +2,16 @@ linters: linters_with_tags(
     tags = NULL, # include all linters
     implicit_integer_linter = NULL,
     extraction_operator_linter = NULL,
-    undesirable_function_linter = NULL,
+    keyword_quote_linter = NULL,
+    undesirable_function_linter(
+      fun = c(
+         # Base messaging
+         "message" = "use cli::cli_inform()",
```
Comment (seabbs) [inline .lintr:9]: really good idea. We could do with this in epinowcast

File: DESCRIPTION [hunk 2]
```diff
@@ -148,7 +149,7 @@ Encoding: UTF-8
 Language: en-GB
 LazyData: true
 Roxygen: list(markdown = TRUE)
-RoxygenNote: 7.3.2
+RoxygenNote: 7.3.2.9000
```
Comment (seabbs) [inline DESCRIPTION:152]: this might make all the cli unhappy?

File: NEWS.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 ## Model changes
 
+- All functions now use the `{cli}` R package to signal errors, warnings, and messages. By @jamesmbaazam in #762 and reviewed by @<REVIEWER>.
```
Comment (seabbs) [inline NEWS.md:11]:
```suggestion
- All functions now use the `{cli}` R package to signal errors, warnings, and messages. By @jamesmbaazam in #762 and reviewed by @seabbs.
```

**Review by seabbs — CHANGES_REQUESTED** [review]
This looks great. Like the lintr as the package check and the use of colour highlighting. You have some very unhappy CI which I expect is due to error and warning testing needing to be updated.

**Comment by jamesmbaazam** [comment]
Sorry, I should have marked this as a draft. Have a few simple fixes then we can try the new changes and see if happy before merging, or we can merge this and apply fancier stuff in another run. {cli} is addictive and I'm trying not to go crazy.

File: .lintr (hunk 1 above)
Comment (jamesmbaazam) [reply .lintr:9]: Best thing I never knew I could do with `lintr`. It made the refactoring process very smooth. Another good idea for a blog post. Stole the idea from the [pkgdown `.lintr` file](https://github.com/r-lib/pkgdown/blob/819cede3fbbb2410912f4d49365177c9159b809f/.lintr.R), so all credits to the authors there.

**Comment by seabbs** [comment]
> Have a few simple fixes then we can try the new changes and see if happy before merging

Ha I know I think this and do fancier stuff in a new PR

**Review by seabbs — APPROVED** [review]
Nice. Looks good.

**Pull request merged**